src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/ConsumerGroup/RemoveAzureEventHubConsumerGroups.cs

```diff
@@ -14,41 +14,82 @@
+        [Parameter(Mandatory = true, ParameterSetName = ConsumergroupPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "EventHub Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasEventHubName)]
         public string EventHub { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = "ConsumerGroup Name")]
+        [Parameter(Mandatory = true, ParameterSetName = ConsumergroupPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = "ConsumerGroup Name")]
         [ValidateNotNullOrEmpty]
         [Alias(AliasConsumerGroupName)]
         public string Name { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = ConsumergroupInputObjectParameterSet, ValueFromPipeline = true, Position = 0, HelpMessage = "ConsumerGroup Object")]
+        [ValidateNotNullOrEmpty]
+        public PSConsumerGroupAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ConsumergroupResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "ConsumerGroup Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
         public override void ExecuteCmdlet()
         {
+
+            if (ParameterSetName.Equals(ConsumergroupInputObjectParameterSet))
+            {
+                LocalResourceIdentifier identifier = new LocalResourceIdentifier(InputObject.Id);
+
+                ResourceGroupName = identifier.ResourceGroupName;
+                Namespace = identifier.ParentResource;
+                EventHub = identifier.ParentResource1;
+                Name = identifier.ResourceName;
+            }
+            else if (ParameterSetName.Equals(ConsumergroupResourceIdParameterSet))
+            {
+                LocalResourceIdentifier identifier = new LocalResourceIdentifier(ResourceId);
+
+                ResourceGroupName = identifier.ResourceGroupName;
+                Namespace = identifier.ParentResource;
+                EventHub = identifier.ParentResource1;
+                Name = identifier.ResourceName;
+            }
+
+
             // delete a ConsumerGroup 
-            if (ShouldProcess(target:Name, action: string.Format(Resources.RemoveConsumerGroup, Name, EventHub)))
-               {
-                    Client.DeletConsumerGroup(ResourceGroupName, Namespace, EventHub, Name);
-               }
+            if (ShouldProcess(target: Name, action: string.Format(Resources.RemoveConsumerGroup, Name, EventHub)))
+            {
+                Client.DeletConsumerGroup(ResourceGroupName, Namespace, EventHub, Name);
```

Comment:
@v-Ajnava [same comment as the ServiceBus PR](https://github.com/Azure/azure-powershell/pull/6732#discussion_r204582389) -- what was this previously returning? We want to ensure that we are returning the same type as before

Reply:
resolved